docs/source/develop.rst

```diff
@@ -0,0 +1,216 @@
+of bug reports, documentation, code, design proposals, and more.
+This page provides resources to on how best to contribute.
+
+Where to ask for help
+---------------------
+
+Dask conversation happens in the following places:
+
+1.  `StackOverflow #dask tag`_: for usage questions
+2.  `Github Issue Tracker`_: for discussions around new features or established bugs
+3.  `Gitter chat`_: for real-time discussion
+
+For usage questions and bug reports we strongly prefer the use of StackOverflow
+and Github issues over gitter chat.  Github and StackOverflow are more easily
+searchable by future users and so is more efficient for everyone's time.
+Gitter chat is generally reserved for community discussion.
+
+.. _`StackOverflow #dask tag`: http://stackoverflow.com/questions/tagged/dask
+.. _`Github Issue Tracker`: https://github.com/dask/dask/issues/
+.. _`Gitter chat`: https://gitter.im/dask/dask
+
+
+Separate Code Repositories
+--------------------------
+
+Dask maintains code and documentation in a few git repositories hosted on the
+Github ``dask`` organization, http://github.com/dask.  This includes the primary
+repository and several other repositories for different components.  A
+non-exhaustive list follows:
+
+*  http://github.com/dask/dask: The main code repository holding parallel
+   algorithms, the single-machine scheduler, and most documentation.
+*  http://github.com/dask/distributed: The distributed memory scheduler
+*  http://github.com/dask/hdfs3: Hadoop Filesystem interface
+*  http://github.com/dask/s3fs: S3 Filesystem interface
+*  http://github.com/dask/dask-ec2: AWS launching
+*  ...
+
+Git and Github can be challenging at first.  Fortunately good materials exist
+on the internet.  Rather than repeat these materials here we refer you to
+Pandas' documentation and links on this subject at
+http://pandas.pydata.org/pandas-docs/stable/contributing.html
+
+
+Issues
+------
+
+The community discusses and tracks known bugs and potential features in the
+`Github Issue Tracker`_.  If you have a new idea or have identified a bug then
+you should raise it there to start public discussion.
+
+If you are looking for an introductory issue to get started with development
+then check out the `introductory label`_, which contains issues that are good
+for starting developers.  Generally familiarity with Python, NumPy, Pandas, and
+some parallel computing are assumed.
+
+.. _`introductory label`: https://github.com/dask/dask/issues?q=is%3Aissue+is%3Aopen+label%3Aintroductory
+
+
+Development Environment
+-----------------------
+
+Download code
+~~~~~~~~~~~~~
+
+Clone the main dask git repostory (or whatever repository you're working on.)::
+
+   git clone git@github.com:dask/dask.git
+
+
+Install
+~~~~~~~
+
+Install dask and dependencies::
+
+   cd dask
+   pip install -e .[complete]
+
+
+Run Tests
+~~~~~~~~~
+
+Dask uses py.test_ for testing.  You can run tests from the main dask directory
+as follows::
+
+   py.test dask --verbose
```

Comment:
Added a note about installing `py.test` and `moto` in the docs here.  Haven't yet produced a separate requirements file.  Should we?